src/App.tsx

```diff
@@ -15,36 +19,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField  {
+  All = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+};
+
+interface SortOptions {
+  sortField: SortField;
+  isReverse: boolean;
+}
+
+function getSortedGoods(goods: Goods, { sortField, isReverse }: SortOptions) {
+  const sortedGoods = [...goods];
+
+  if (sortField) {
+    sortedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SortField.Alphabet:
+          return a.localeCompare(b);
+
+        case SortField.Length:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = React.useState<SortField>(SortField.All);
+  const [isReverse, setIsReverse] = React.useState(false);
+
+  const sortedGoods: Goods = getSortedGoods(goodsFromServer, {
+    sortField,
+    isReverse,
+  });
+  const isResetButtonVisible = isReverse || sortField;
+
+  const handleReset = () => {
+    setSortField(SortField.All);
+    setIsReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortField.Alphabet,
+          })}
+          onClick={() => {
+            setSortField(SortField.Alphabet);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortField.Length,
+          })}
+          onClick={() => setSortField(SortField.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReverse,
+          })}
+          onClick={() => setIsReverse(isReverseState => !isReverseState)}
+        >
           Reverse
         </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {isResetButtonVisible && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => handleReset()}
```

Comment:
```suggestion
            onClick={handleReset}
```

can be a bit simplify